test/autoCompletion.withschemainfile.test.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { TextDocument } from 'vscode-languageserver';
+import { getLanguageService } from '../src/languageservice/yamlLanguageService';
+import { toFsPath, schemaRequestService, workspaceContext }  from './utils/testHelper';
+import assert = require('assert');
+import path = require('path');
+
+const languageService = getLanguageService(schemaRequestService, workspaceContext, [], null);
+
+const languageSettings = {
+    schemas: [],
+    completion: true
+};
+
+const uri = toFsPath(path.join(__dirname, './fixtures/testArrayMaxProperties.json'));
+languageService.configure(languageSettings);
+
+suite('Auto Completion Tests with schema defined in file', () => {
+
+    describe('yamlCompletion with schema defined in file', function () {
+
+            function setup(content: string) {
+                return TextDocument.create('file://~/Desktop/vscode-k8s/test-schemainfile.yaml', 'yaml', 0, content);
+            }
+
+            function parseSetup(content: string, position) {
+                const testTextDocument = setup(content);
+                return languageService.doComplete(testTextDocument, testTextDocument.positionAt(position), false);
+            }
+
+            it('Provide completion from schema declared in file', done => {
+                const content = `# yaml-language-server: $schema=${uri}\n- `;
+                const completion = parseSetup(content, content.length);
+                completion.then(function (result) {
+                    assert.equal(result.items.length, 3);
+                }).then(done, done);
+            });
+            
+            it('Provide completion from schema declared in file with several attributes', done => {
```

Comment:
Is there any way you can rebase on master and integrate these tests into the main autoCompletion.test.ts file? And also check the actual values of result.items. Also can you add another test for completion with yaml-language-server modelline in multiple documents. I've been trying to harden the tests lately

Reply:
I added a test with multiple documents. it seems there is no "inserttext" which is added:
(node:25674) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: Expected values to be loosely deep-equal:

```
{
  documentation: '',
  insertText: 'prop1: $1',
  insertTextFormat: 2,
  kind: 10,
  label: 'prop1'
}

should equal

{
  documentation: '',
  insertText: 'prop1: $1',
  insertTextFormat: 2,
  kind: 10,
  label: 'prop1',
  textEdit: {
    newText: 'prop1: $1',
    range: {
      end: {
        character: 2,
        line: 0
      },
      start: {
        character: 2,
        line: 0
      }
    }
  }
}
```

I don't get why it woudl be specific to what I changed. I will have a look tomorrow.

Reply:
this is an existing bug on master https://github.com/redhat-developer/yaml-language-server/issues/288

I think we can go without checking for the specific completion in this PR.